specs/flyout/flyout.md

```diff
@@ -0,0 +1,179 @@
+
+### Use Cases
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item.
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Popover target element. This way it can be dynamic and adapt to the available space.
+
+**Soft Dismiss (aka Light Dismiss):** - A way to quickly close the Popover by clicking outside it, or pressing the `esc` key. An event would be emitted anytime the Popover is dismissed. The Popover would always be soft-dismissable. Detecting the users next click of a different element can trigger this when clicking outside the Popover.
+
+**Focus Management:** - Should follow the same process for focus management as a Dialog, primarily as the Popover's role will be 'dialog'.
+
+This includes:
+- Trap focus - optional, defaults to enabled - Focus is contained within the popover while it is visible.
+- First element focus - Focusing on the first focusable element when trap focus is enabled, but not overriding an element that is set to be focused by the author.
+- Returning focus to the target - optional, defaults to enabled, after popover is closed the focus would return to the target element.
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-popover`
+
+*Attributes*
+- `visible` - boolean, whether or not the Popover is visible, defaults to undefined, as this is controlled by the author
+- `trap-focus` - boolean, whether or not to keep focus contained inside the popover, defaults to true
+- `target` - string, the html id of the HTMLElement that the Popover is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `focus-target-on-close` - boolean, whether or not to focus back on to the _target element_ when the popover is closed, defaults to true
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic=
```

Comment:
This should describe all valid values for `position`, or link to the Anchored Region documentation for `position` (or both).

Reply:
+1 - Should we support the same positioning values as tooltip? Less? More?

Reply:
I have updated with a note to see the _Position Options_ section of this spec, as it has several options available.